frontend/renderer/src/particle_renderer.rs

```diff
@@ -164,7 +246,9 @@ void main() {
         let attribs: Vec<Particle> = self
             .particles
             .iter()
-            .filter(|x| x.creation_time >= current_time - 10.0)
+            .filter(|x| {
+                x.creation_time <= current_time && x.creation_time != EMPTY_PARTICLE_CREATION_TIME
+            })
```

Comment:
I think this logic was flipped prior to this.

Reply:
How about `current_time >= x.creation_time && current_time <= x.creation_time + x.lifetime`